src/main/java/net/tridentsdk/chat/ChatComponent.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Trident - A Multithreaded Server Alternative
+ * Copyright 2016 The TridentSDK Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.tridentsdk.chat;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import lombok.*;
+import lombok.experimental.Accessors;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Accessors(chain = true)
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class ChatComponent {
+
+    @Getter @Setter
+    private String text, translate, selector, insertion, scoreUsername, scoreObjective;
+
+    @Getter @Setter
+    private ChatColor color;
+
+    @Getter @Setter
+    private ClickEvent clickEvent;
+
+    @Getter @Setter
+    private HoverEvent hoverEvent;
+
+    private List<JsonElement> with = new LinkedList<>();
+    private List<ChatComponent> extra = new LinkedList<>();
+    private List<String> extraStrings = new LinkedList<>();
+    private AtomicBoolean bold, italic, underlined, strikethrough, obfuscated;
```

Comment:
Why are these boolean values atomic if none of the other fields are thread safe? If ChatComponent is intended to be used across multiple threads, then the other values such as the lists and strings should be thread-safe as well.